docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+
+First, create your HTML so that appears somewhere within the Wagtail admin.
+
+```html
+<!-- Will log 'My controller has connected: hi' to the console -->
+<div data-controller="my-controller">Hi</div>
+<!-- Will log 'My controller has connected: hello' to the console, with the span element-->
+<div data-controller="my-controller">
+    Hello
+    <span data-my-controller-target="label"></span>
+</div>
+```
+
+Second, create a JavaScript file that will contain your Controller code.
+This Controller logs a simple message on `connect`, which is once the Controller has been created and connected to a HTML element with the matching `data-controller` attribute.
+
+```javascript
+// myapp/static/js/example.js
+
+// 1. With classes
+(() => {
+    class MyController extends window.Stimulus.use.Controller {
+        static targets = ['label'];
+        connect() {
+            console.log(
+                'My controller has connected:',
+                this.element.innerText,
+                this.labelTargets,
+            );
+        }
+    }
+
+    window.Stimulus.register('my-controller', MyController);
+})();
+
+// OR 2. With Objects
+(() => {
+    const MyController = {
+        STATIC: { targets: ['label'] },
+        connect() {
+            console.log(
+                'My controller has connected:',
+                this.element.innerText,
+                this.labelTargets,
+            );
+        },
+    };
+
+    window.Stimulus.register(
+        'my-controller',
+        window.Stimulus.use.createController(MyController),
+    );
+})();
+```
+
+Third, load the JavaScript file into Wagtail's admin with a hook.
+
+```python
+# myapp/wagtail_hooks.py
+from django.templatetags.static import static
+from django.utils.safestring import mark_safe
+
+from wagtail import hooks
+
+@hooks.register('insert_global_admin_js')
+def global_admin_js():
+    return mark_safe(
+        f'<script src="{static("js/example.js")}"></script>',
+    )
+```
+
+You should now be able to refresh your admin page that was showing the HTML and see two logs in the console.
+
+#### A word count controller example without a build system
+
+This is a more complex example, we will create a Controller that adds a small `output` element next to the controlled `input` element that shows a count of how many words have been typed.
+
+```javascript
+// myapp/static/js/word-count-controller.js
+// By passing the `window.Stimulus` to the immediate function call we can have simpler code
+((Stimulus) => {
+    class WordCountController extends Stimulus.use.Controller {
+        static values = { max: { default: 10, type: Number } };
+
+        connect() {
+            this.setupOutput();
+            this.updateCount();
+        }
+
+        setupOutput() {
+            if (this.output) return;
+            const template = document.createElement('template');
+            template.innerHTML = `<output name='word-count' for='${this.element.id}' style='float: right;'></output>`;
```

Comment:
I think we should avoid inline styles in example code in the spirit of CSP compatibility.